src/main/java/seedu/address/model/appointment/Appointment.java

```diff
@@ -0,0 +1,126 @@
+    private final Person patient;
+    private final String description;
+    // private final List<Medicine> medicinePackingList;
+
+    public Appointment(DateTime startDateTime, DateTime endDateTime, RecurringDateTime frequency,
+                       Person patient, String description) {
+        requireAllNonNull(startDateTime, endDateTime, frequency, patient, description);
+        this.startDateTime = startDateTime;
+        this.endDateTime = endDateTime;
+        this.frequency = frequency;
+        this.patient = patient;
+        this.description = description;
+    }
+
+    public DateTime getStartDateTime() {
+        return startDateTime;
+    }
+
+    public DateTime getEndDateTime() {
+        return endDateTime;
+    }
+
+    public RecurringDateTime getFrequency() {
+        return frequency;
+    }
+
+    public Person getPatient() {
+        return patient;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public Address getLocation() {
+        return patient.getAddress();
+    }
+
+    /**
+     * Returns true if both appointments of the same date and time have at least one other identity field that is the
+     * same.
+     * This defines a weaker notion of equality between two appointments.
+     */
+    public boolean isSameAppointment(Appointment otherAppointment) {
+        if (otherAppointment == this) {
+            return true;
+        }
+
+        return otherAppointment != null
+                && otherAppointment.getStartDateTime().equals(getStartDateTime())
+                && (otherAppointment.getEndDateTime().equals(getEndDateTime())
+                    || otherAppointment.getFrequency().equals(getFrequency()));
+    }
+
+    /**
+     * Returns true if both appointments have the same identity and data fields.
+     * This defines a stronger notion of equality between two appointments.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Appointment)) {
+            return false;
+        }
+
+        Appointment otherAppointment = (Appointment) other;
+
+        return otherAppointment.getStartDateTime().equals(getStartDateTime())
+                && otherAppointment.getEndDateTime().equals(getEndDateTime())
+                && otherAppointment.getFrequency().equals(getFrequency())
+                && otherAppointment.getPatient().equals(getPatient())
+                && otherAppointment.getDescription().equals(getDescription())
+                && otherAppointment.getLocation().equals(getLocation());
+    }
+
+    @Override
+    public int hashCode() {
+        // use this method for custom fields hashing instead of implementing your own
+        return Objects.hash(startDateTime, endDateTime, frequency, patient, description);
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
```

Comment:
Does this need to be final?

Reply:
Again, this was to stay consistent with the way the Person class had their toString() method implemented. I think the reason for wanting the reference to the string printed to be immutable is to prevent any assignment to `null` or something in other parts of the code.